cmd/shortener/main.go

```diff
@@ -1,57 +1,92 @@
-	gziphandle "github.com/DaniYer/GoProject.git/internal/app/gzipmiddleware"
+	"github.com/DaniYer/GoProject.git/internal/app/gzipmiddleware"
 	"github.com/DaniYer/GoProject.git/internal/app/logging"
+	"github.com/DaniYer/GoProject.git/internal/app/ping"
 	"github.com/DaniYer/GoProject.git/internal/app/redirect"
 	"github.com/DaniYer/GoProject.git/internal/app/shortener"
-	"github.com/DaniYer/GoProject.git/internal/app/storage"
+	"github.com/DaniYer/GoProject.git/internal/app/storage/database"
+	"github.com/DaniYer/GoProject.git/internal/app/storage/file"
+	"github.com/DaniYer/GoProject.git/internal/app/storage/memory"
 	"github.com/go-chi/chi/v5"
-
+	_ "github.com/jackc/pgx/v5/stdlib"
 	"go.uber.org/zap"
 )
 
-var (
-	sugar *zap.SugaredLogger
-	cfg   = config.ConfigInit()
-)
-
 func main() {
-	router := chi.NewRouter()
+	// Инициализация конфигурации
+	cfg := config.NewConfig()
 
+	// Логгер
 	logger, err := zap.NewDevelopment()
 	if err != nil {
 		panic(err)
 	}
 	defer logger.Sync()
-
-	// Создаём SugaredLogger и передаём его в logging.InitLogger()
-	sugar = logger.Sugar()
+	sugar := logger.Sugar()
 	logging.InitLogger(sugar)
 
-	sugar.Infow(
-		"Starting server",
-		"addr", cfg.A,
+	var (
+		db          *sql.DB
+		store       shortener.URLStore
+		storeWithDB shortener.URLStoreWithDBforHandler
 	)
-	write, _ := storage.NewFileStorage(cfg.F)
-	read, _ := storage.NewConsumer(cfg.F)
-
-	router.Use(logging.WithLogging) // Теперь логгер в logging будет работать
-	router.Use(gziphandle.GzipHandle)
-	router.Post("/", func(w http.ResponseWriter, r *http.Request) {
-		shortener.GenerateShortURLHandler(w, r, cfg, write)
-	})
-	router.Get("/{id}", func(w http.ResponseWriter, r *http.Request) {
-		redirect.RedirectToOriginalURL(w, r, read)
-	})
-	router.Post("/api/shorten", func(w http.ResponseWriter, r *http.Request) {
-		shortener.HandleShortenURL(w, r, cfg, write)
-	})
-
-	if err := http.ListenAndServe(cfg.A, router); err != nil {
-		sugar.Errorf("RIP %v", err) // исправлено форматирование ошибки
+
+	// Выбор хранилища (БД → файл → память)
+	if cfg.DatabaseDSN != "" && cfg.DatabaseDSN != "localDB" {
+		db, err = database.InitDB("pgx", cfg.DatabaseDSN)
+		if err != nil {
+			sugar.Errorf("Ошибка подключения к БД: %v", err)
+		} else {
+			if err := database.CreateTable(db); err != nil {
+				sugar.Errorf("Ошибка создания таблицы: %v", err)
+			} else {
+				dbStore := database.NewDBStore(db)
+				store = dbStore
+				storeWithDB = dbStore
+			}
+		}
+	}
+
+	if store == nil && cfg.FileStoragePath != "" {
+		fs, err := file.NewFileStore(cfg.FileStoragePath)
+		if err != nil {
+			sugar.Errorf("Ошибка инициализации файлового хранилища: %v", err)
+		} else {
+			store = fs
+			storeWithDB = fs
+		}
+	}
+
+	if store == nil {
+		sugar.Infof("Используется in-memory хранилище")
+		memStore := memory.NewMemoryStore()
+		store = memStore
+		storeWithDB = memStore
+	}
+
+	// Роутер
+	router := chi.NewRouter()
```

Comment:
Please simplify the `main` package.
You can create a function such as `InitializeApp`.

P.S. Please look at it.
https://github.com/google/wire/tree/main/_tutorial